Bugfix FXIOS-10491 - [Unified Search] The search engine selection bottom sheet is hidden by an open keyboard in iPhone landscape

PARAIPAN9 opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10491)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22978)

## :bulb: Description
<!--- Describe your changes so the reviewer can understand the context and decisions made for your work -->
- Cancels edit mode when search engine vc is a page sheet to avoid overlapping when rotating in landscape.
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

thatswinnie commented on firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift:
```diff
@@ -589,6 +589,7 @@ class BrowserCoordinator: BaseCoordinator,
             navigationController.modalPresentationStyle = .pageSheet
             navigationController.sheetPresentationController?.detents = [.medium(), .large()]
             navigationController.sheetPresentationController?.prefersGrabberVisible = true
+            store.dispatch(ToolbarAction(windowUUID: windowUUID, actionType: ToolbarActionType.cancelEdit))
```
I don't think this is the best solution from a usability perspective: imagine the user has typed their search term already and then wants to switch search engines. Now with us canceling edit mode they lost everything they typed and have to start over entering the search term. 
I think it would be better to just dismiss the keyboard but stay in edit mode.

PARAIPAN9 replied:
Yes, you're right. I believe that that was the behavior before, but by rotating the device while in edit mode, triggers the keyboard.

ih-codes replied:
Ahhh, so THAT is what's triggering the keyboard to reappear? 🤔  Good catch @PARAIPAN9 , thanks for looking into this for me! 🙏 

@PARAIPAN9 If we can't use cancelEdit, maybe we need to do something similar as AddressBarState's `isScrollingDuringEdit`, which when set to true hides the keyboard on line 268 of LocationView:

```
        let shouldShowKeyboard = state.isEditing && !state.isScrollingDuringEdit
```

So maybe we can add another state property so `shouldShowKeyboard` returns false (or at least doesn't get reset to true) when we rotate AND the search engine bottom sheet is visible. Possibly we need to set something on the redux state every time the search engine bottom sheet is shown, and then clear it when it is dismissed. 🧐

PARAIPAN9 replied:
Yes @ih-codes, this seems like a good approach.

PARAIPAN9 commented:
I've considered that it is redundant to add another property that does exactly the same thing as the `isScrollingDuringEdit` property, so I just renamed it to better fit its purpose. I've also renamed the action type.

ih-codes commented:
> I've considered that it is redundant to add another property that does exactly the same thing as the `isScrollingDuringEdit` property, so I just renamed it to better fit its purpose. I've also renamed the action type.

Great choice @PARAIPAN9, this is fantastic! Thank you for your help! 🙏

PARAIPAN9 commented:
@ih-codes, you're welcome!!
